### DISCUSSION TITLE
parquet files glob limit/recommendations

### DISCUSSION THREAD

**Discussion by darked89** [duckdb/duckdb#3117] [Q&A] [answered]
Hello,

Is there some hard limit of number of parquet files for doing:
```
SELECT * FROM 'inputs/*.parquet' WHERE contig='contig_01'; 
```
using a duckdb Python library?

I have in mind something around 5-6k parquet files (sorted by contig column and with exactly the same set of 50 contigs => partitioned data set). 

Assuming that such SELECT is doable, are there any recommendations to make it faster/robust?

Many thanks for your help,

Darek Kedra

**Comment by Mytherin** [comment, marked as answer]
There is no built-in limit on the number of Parquet files that can be read. You may want to enable metadata caching if you intend to run multiple queries on the same data set (`PRAGMA enable_object_cache`), since metadata reading can be quite costly when many Parquet files are involved.

**Reply by darked89** [reply]
Thank you. I will use ```PRAGMA``` as advised.

DK
